query-coordinator/src/main/scala/com/socrata/querycoordinator/QueryRewriter.scala

```diff
@@ -260,6 +260,54 @@ class QueryRewriter(analyzer: SoQLAnalyzer[SoQLType]) {
     }
   }
 
+  def isLiteral(expr: Expr): Boolean = {
+    expr match {
+      case _: ColumnRef =>
+        false
+      case _: TypedLiteral[_] =>
+        true
+      case fc: FunctionCall =>
+        !fc.function.isAggregate &&
+          fc.parameters.nonEmpty &&
+          fc.parameters.forall(x => isLiteral(x))
+      case _ =>
+        false
+    }
+  }
+
+  private def rewriteDateTrunc(r: Anal, rollupColIdx: Map[Expr, Int], fc: FunctionCall): Option[Expr] = {
+    val fn = fc.function.function
+    val dateTrunFn: Option[Function[SoQLType]] = dateTruncHierarchy.find(f => f == fn)
+    dateTrunFn.flatMap { _ =>
+      rollupColIdx.get(fc) match {
+        case Some(ruColIdx) =>
+          Some(typed.ColumnRef(NoQualifier, rollupColumnId(ruColIdx), SoQLFloatingTimestamp.t)(fc.position))
+        case None =>
+          val left +: _ = fc.parameters
```

Comment:
`fc.head` is better than pattern-matching on a destructuring `+:` and discarding the second half.  Also I'm not _super_ happy with passing in a `FunctionCall` and having it implicit that there'll be at least one parameter, but I honestly don't see a better way.

Reply:
..that is, `fc.parameters.head`